### ISSUE TITLE
cli bad option: --inspect --max-old-space-size=4096

### ISSUE THREAD

**Issue by stoically** [mochajs/mocha#3853]
### Prerequisites

- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [x] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend that you _not_ install Mocha globally.

### Description

Executing of `mocha --inspect --max-old-space-size=4096` results in the cli responding with a `bad option` error.

### Steps to Reproduce

See description.

**Expected behavior:** [What you expect to happen]

Should not result in `bad option`.

**Actual behavior:** [What actually happens]

Results in `bad option`.

**Reproduces how often:** [What percentage of the time does it reproduce?]

100%

### Versions

- The output of `mocha --version` and `node node_modules/.bin/mocha --version`: 6.0.2
- The output of `node --version`: v10.15.3
- Your operating system
  - name and version: Ubuntu 18.04 LTS
  - architecture (32 or 64-bit): 64bit
- Your shell (e.g., bash, zsh, PowerShell, cmd): bash
- Your browser and version (if running browser tests):
- Any third-party Mocha-related modules (and their versions):
- Any code transpiler (e.g., TypeScript, CoffeeScript, Babel) being used (and its version): none

**Comment by plroebuck** [comment]
Believe this was already fixed in as-yet-unreleased PR #3788.
The code from the routine below was incorrectly combining multiple arguments into a single string.

You can patch your version of Mocha by changing `unparseNodeFlags` in **"lib/cli/node-flags.js"** to be:
```js
exports.unparseNodeFlags = opts => {
  var args = unparse(opts);
  return args.length
    ? args
        .join(' ')
        .split(/\b/)
        .map(arg => (arg === ' ' ? '=' : arg))
        .join('')
        .split(' ')        // <- add this line
    : [];
};
```

**Comment by plroebuck** [comment]
Closing this issue as already fixed, to be released in Mocha-6.1.
If the above patch does not address your problem, reply as such and I'll reopen this.

**Comment by stoically** [comment]
Can confirm, it works with #3787 in place. Thanks!
